notify/util.go

```diff
@@ -167,6 +176,20 @@ func readAll(r io.Reader) string {
 	return string(bs)
 }
 
+// getFailureStatusCodeCategory return the status code category for failure request
+// the status starts with 4 will return 4xx and starts with 5 will return 5xx
+// other than 4xx and 5xx input status will return an error.
+func getFailureStatusCodeCategory(statusCode int) (string, error) {
```

Comment:
Based on how we use this, I feel we should not return error here, but just return default value, WDYT? Would make more clean and simple.

Reply:
I kind of wanting to return an error so that the user of the function has knowledge about if the status code has been successfully translated. Otherwise they will have to rely on comparing if the result = failureUnknownCategoryCode if failureUnknownCategoryCode is not the ideal result for them in error case. WDYT?

Reply:
Hm, well there are no users of this function yet, and as YAGNI goes, we should assume there will be none. I would focus on our usage:

```
result, interErr := getFailureStatusCodeCategory(e.StatusCode)
			if interErr == nil {
				statusCodeCategory = result
			}
```

We don't want to know IF the status was translated correctly, but rather we want `defaultStatusCodeCategory` in this case. So I would just return `defaultStatusCodeCategory` 

Also, now when we talk about it, `util` is not a great place for it - it's pretty generic - we want it in util if we have more than one usage of it. Given the function is very small and used only in one place, why not:

1. Moving this function next to `notify.go` file? 
OR
2. Copy the content of this function to caller place? It's might be too shallow for a function, but up to you (: